test-builder/api.ts

```diff
@@ -441,30 +449,27 @@ const buildIDLMemberTests = async (
 
   for (const member of members) {
     const isStatic = member.special === "static" || iface.type === "namespace";
-    const name = member.name + (isStatic ? "_static" : "");
-
-    if (handledMemberNames.has(name)) {
-      continue;
-    }
-
     const isEventHandler =
       member.idlType?.type === "attribute-type" &&
       typeof member.idlType?.idlType === "string" &&
       member.idlType?.idlType.endsWith("EventHandler");
 
-    if (isEventHandler) {
-      // XXX Tests for events will be added with another package, see
-      // https://github.com/openwebdocs/mdn-bcd-collector/issues/133 for
-      // details. In the meantime, ignore event handlers.
+    const name = isEventHandler
+      ? `${member.name.replace(/^on/, "")}_event`
+      : member.name + (isStatic ? "_static" : "");
+
+    if (handledMemberNames.has(name)) {
       continue;
     }
 
     let expr: string | RawTestCodeExpr | RawTestCodeExpr[] = "";
 
-    // Constructors, constants, and static attributes should not have
+    // Constructors, constants, event handlers and static attributes should not have
     // auto-generated custom tests
     const customTestExactMatchNeeded =
-      isStatic || ["toString", "toJSON"].includes(member.name as string);
+      isStatic ||
+      isEventHandler ||
+      ["toString", "toJSON"].includes(member.name as string);
```

Comment:
I think that auto-generating event handler tests from custom tests is actually what we want, as this ensures that we're using an instance when needed.

Reply:
If I leave this in, the generated test is what we want, I think:

```
"XMLHttpRequestEventTarget" in self && "onabort" in XMLHttpRequestEventTarget.prototype
```


If revert this, I the generated test is:

```
(function () {
  if (!("XMLHttpRequest" in self)) {
    return { result: false, message: "XMLHttpRequest is not defined" };
  }
  var instance = new XMLHttpRequest();
  return !!instance && "abort_event" in instance;
})();
```

`"abort_event" in instance` doesn't work of course, so I thought I needed to add `isEventHandler` to `customTestExactMatchNeeded`.

Thoughts?

Reply:
ahh thank you! :)